File: app/src/main/java/com/dominikgaller/alpinebooking/guide/core/domain/guidetour/GuideTourId.java
package com.dominikgaller.alpinebooking.guide.core.domain.guidetour;

import java.util.Objects;
import java.util.UUID;

/**
 * Identity value object for a {@link GuideTour} aggregate.
 *
 * <p>Wraps a {@link UUID}. Equality is by value (record semantics).
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}.
 */
public record GuideTourId(UUID value) {

    public GuideTourId {
        Objects.requireNonNull(value, "GuideTourId value must not be null");
    }

    public static GuideTourId generate() {
        return new GuideTourId(UUID.randomUUID());
    }
}

File: src/main/kotlin/com/example/contractmanagement/guide/core/domain/guidetour/GuideTourId.kt
package com.example.contractmanagement.guide.core.domain.guidetour

import java.util.UUID

/**
 * Identity of a [GuideTour].
 *
 * Owned by the `guide` context. The `booking` context receives it as an opaque [String] on
 * a domain event and never parses it — see
 * `documentation/adr/0005-bounded-context-identity-boundaries.adr.md`.
 *
 * SDD: see `documentation/domain/aggregate-guide-tour.spec.md`.
 */
data class GuideTourId(
    val value: UUID,
) {
    companion object {
        fun generate(): GuideTourId = GuideTourId(UUID.randomUUID())
    }
}
